content/en/platform/corda/4.12/community/comm-upgrade-guide.md

```diff
@@ -0,0 +1,275 @@
+* Keep a copy of the old CorDapp contract JAR file in the new `legacy-contracts` folder placed inside all upgraded 4.12 nodes.
+
+Any future Corda 4.12 nodes added to the network will also require the `legacy-contracts` folder, unless all nodes have been updated by that time. In the latter case, you are adding new Corda 4.12 nodes to a non-mixed network and this requirement is no longer necessary.
+
+### Upgrading all network nodes to 4.12
+
+In this scenario, you are creating a non-mixed network composed solely of nodes operating on Corda 4.12. Transactions occurring within this newly-upgraded network only have the 4.12 contract component group and therefore, you do not need the `legacy-contracts` folder. However, it is still important that you keep a copy of the old contract JAR for reference.
+
+If any new Corda 4.12 nodes are added to this network in the future, a problem arises wherein new nodes won’t trust old contract attachments. For example, this happens if a new transaction contains a backchain of old Corda 4.x transactions. For this reason, you must upload the old contract JAR via RPC to the new node, ensuring it trusts the old contract version.
+
+### Adding 4.12 nodes to a new network
+
+This scenario is the simplest, as it does not require any complex update procedures. In this scenario, all CorDapps are developed for Java 17 and Kotlin 1.9, and the process for adding new nodes is the same as setting up a new network of any previous Corda version.
+
+## Upgrade prerequisites
+
+This guide assumes that you have a working Corda network with one or more Corda 4.11 nodes running. If you have custom CorDapps running on your nodes, this guide also describes the required upgrade steps for CorDapps from version 4.11 to 4.12.
+
+To complete the upgrade from Corda 4.11 to Corda 4.12, you need the following components:
+* A Corda 4.12 JAR (obtained from R3).
+* A Corda Transaction Validator Utility JAR (obtained from R3, as part of the Corda 4.12 release package).
+* Access to, and the ability to edit the source code for any existing custom CorDapps running in the nodes to be upgraded from 4.11 to 4.12.
+
+## Upgrade outline
+
+To upgrade your Corda node from version 4.11 to 4.12, you must perform the following steps:
+
+1. Validate all existing transactions (per node). See [Validate transactions]({{< relref "#validate-transactions" >}}).
+2. Upgrade any custom CorDapps running on the Corda 4.11 node to work with Java 17 and Kotlin 1.9. See [Upgrade 4.11 CorDapps]({{< relref "#upgrade-411-cordapps" >}}).
+3. Preserve old CorDapp contracts in a new folder called `legacy-contracts`. See [Add the legacy contracts folder to mixed networks]({{< relref "#add-the-legacy-contracts-folder-to-mixed-networks" >}}).
+   {{< note >}}
+   This step is for mixed networks only. It is not required if you plan on upgrading all nodes on your network to 4.12.
+   {{< /note >}}
+
+### Validate transactions
```

Comment:
@chriscochrane Should all this section be removed? It's all about using TVU. Pls confirm.

Reply:
Yes it needs to be removed, but we need to leave something in that says the user is responsible for verifying that their existing txn backchains will work with 4.12, but I guess without specifying how they should go about doing that.

Reply:
Removed